### PR TITLE
[test] Add workaround for PGI on Tsa

### PR DESCRIPTION
**Pull request by ajocksch**
This change introduces the workaround

-D__PGIC__=19

### CODE REVIEW

**Review by kraushm — APPROVED** [review]
lgtm.

**Review by victorusu — APPROVED** [review]
`lgtm`

File: cscs-checks/mch/gpu_direct_cuda_tsa_pgi_workaround.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        self.num_gpus_per_node = 1
+        self.num_tasks_per_node = 1
+        result = sn.extractsingle(r'Result :\s+(?P<result>\d+\.?\d*)',
+                                  self.stdout, 'result', float)
+        self.sanity_patterns = sn.assert_reference(result, 1., -1e-5, 1e-5)
+        self.maintainers = ['AJ', 'MKr']
+        self.tags = {'production', 'mch', 'craype'}
+
+    @rfm.run_before('compile')
+    def setflags(self):
```
Comment (vkarak) [inline cscs-checks/mch/gpu_direct_cuda_tsa_pgi_workaround.py:41]: As discussed offline, please add a meaningful name here and a meaningful PR title.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/mch/gpu_direct_cuda_tsa_pgi_workaround.py (hunk 1 above)
Comment (vkarak) [reply cscs-checks/mch/gpu_direct_cuda_tsa_pgi_workaround.py:41]:
```suggestion
    def pgi_workaround_tsa(self):
```

**Review by vkarak — CHANGES_REQUESTED** [review]
Wait a sec, why do we need a new test for this?! Wouldn't just a hook to the existing one work?

**Review by vkarak — APPROVED** [review]

**Pull request merged**